src/mbgl/layout/symbol_instance.hpp

```diff
@@ -28,6 +28,7 @@ class SymbolInstance {
     CollisionFeature textCollisionFeature;
     CollisionFeature iconCollisionFeature;
     WritingModeType writingModes;
+    std::size_t featureIndex;
```

Comment:
I think this is okay for now. I started learning more about the various classes used during symbol placement and I'm thinking maybe we can move to a hierarchy where `SymbolFeature` owns `SymbolInstance`s, and can provide a direct `SymbolFeature&` reference at the appropriate time.